### PR TITLE
Support separate dev/prod metrics logging.

### PR DESCRIPTION
**Pull request by rockapot**

### CODE REVIEW

File: Sources/PromotedAIMetricsSDK/NetworkConnection.swift [hunk 1]
```diff
@@ -34,12 +33,25 @@ public protocol NetworkConnection {
...
   ///   are passed through `callback`.
-  func sendMessage(_ message: Message, url: URL, clientConfig: ClientConfig,
+  func sendMessage(_ message: Message, clientConfig: ClientConfig,
                    callback: Callback?) throws
 }
 
 // MARK: -
 extension NetworkConnection {
+  func metricsLoggingURL(clientConfig: ClientConfig) throws -> URL {
+    #if DEBUG
```
Comment (danbosnichill) [inline Sources/PromotedAIMetricsSDK/NetworkConnection.swift:43]: Is it a common practice for local workstations and test/dev environments to use DEBUG?

File: Sources/PromotedAIMetricsSDK/ClientConfig/ClientConfig.swift [hunk 2]
```diff
@@ -11,8 +12,10 @@ public class ClientConfig: NSObject {
   var loggingEnabled: Bool = true
   
   @objc public var metricsLoggingURL: String = ""
+  @objc public var devMetricsLoggingURL: String = ""
```
Comment (danbosnichill) [inline Sources/PromotedAIMetricsSDK/ClientConfig/ClientConfig.swift:15]: For web, we'll probably have different environment params for each of these that fill in the same values for dev vs prod depending on the case.  So dev would reuse `metricsLoggingURL` but have different values.  I can see this changing then when we're further along and use a dynamic config solution.

File: Sources/PromotedAIMetricsSDK/NetworkConnection.swift (hunk 1 above)
Comment (rockapot) [reply Sources/PromotedAIMetricsSDK/NetworkConnection.swift:43]: There's no single convention about local build vs. prod build. I considered a few things:

1. Checking if we're running in the iOS simulator (this misses if we're running locally on device).
2. Checking if we're running on an App Store build (this is somewhat complex and probably more trouble than it's worth).
3. Checking if we're running in debug.

Given that, when people are running locally, it's almost always for debug/development, choice 3 seemed safest.

File: Sources/PromotedAIMetricsSDK/ClientConfig/ClientConfig.swift (hunk 2 above)
Comment (rockapot) [reply Sources/PromotedAIMetricsSDK/ClientConfig/ClientConfig.swift:15]: We can replicate this behavior by duplicating the URL but with different params. Let me know how this shapes up on the server side and we can update this.

**Pull request merged**